llama-index-integrations/embeddings/llama-index-embeddings-nomic-multimodal/llama_index/embeddings/nomic_multimodal/base.py

```diff
@@ -0,0 +1,127 @@
+from enum import Enum
+from typing import List, Optional
+
+import nomic
+import nomic.embed
+import warnings
+from PIL import Image
+from llama_index.core.bridge.pydantic import Field
+from llama_index.core.callbacks import CallbackManager
+from llama_index.core.embeddings.multi_modal_base import MultiModalEmbedding
+from llama_index.core.schema import ImageType
+
+
+class NomicTaskType(str, Enum):
+    SEARCH_QUERY = "search_query"
+    SEARCH_DOCUMENT = "search_document"
+    CLUSTERING = "clustering"
+    CLASSIFICATION = "classification"
+
+
+class NomicInferenceMode(str, Enum):
+    REMOTE = "remote"
+    LOCAL = "local"
+    DYNAMIC = "dynamic"
+
+
+class NomicMultiModalEmbedding(MultiModalEmbedding):
```

Comment:
Do you think its possible to, instead of adding a new class, make the existing nomic embeddings extend MultiModalEmbedding?

Reply:
I think so. Let me update.